erizo_controller/erizoClient/src/webrtc-stacks/BaseStack.js

```diff
@@ -148,130 +117,337 @@ const BaseStack = (specInput) => {
-    Logger.debug('processAnswer - Remote Description', msg.type, msg.sdp);
-    Logger.debug('processAnswer - Local Description', msg.type, localDesc.sdp);
-    that.remoteSdp = remoteSdp;
-
-    remoteDesc = msg;
-    that.peerConnection.setLocalDescription(localDesc)
-      .then(() => that.peerConnection.setRemoteDescription(new RTCSessionDescription(msg)))
-      .then(() => {
-        specBase.remoteDescriptionSet = true;
-        Logger.info('Candidates to be added: ', specBase.remoteCandidates.length,
-                      specBase.remoteCandidates);
-        while (specBase.remoteCandidates.length > 0) {
-          // IMPORTANT: preserve ordering of candidates
-          that.peerConnection.addIceCandidate(specBase.remoteCandidates.shift());
-        }
-        Logger.info('Local candidates to send:', specBase.localCandidates.length);
-        while (specBase.localCandidates.length > 0) {
-          // IMPORTANT: preserve ordering of candidates
-          specBase.callback({ type: 'candidate', candidate: specBase.localCandidates.shift() });
-        }
-      })
-      .catch(errorCallback.bind(null, 'processAnswer', undefined))
-      .then(() => {
-        firstLocalDescriptionSet = true;
-        firstLocalDescriptionQueue.stopEnqueuing();
-        firstLocalDescriptionQueue.dequeueAll();
+      msg.sdp = remoteSdp.toString();
+
+      configureLocalSdpAsOffer();
+
+      Logger.debug('processAnswer - Remote Description', msg.type, msg.sdp);
+      Logger.debug('processAnswer - Local Description', msg.type, localDesc.sdp);
+      that.remoteSdp = remoteSdp;
+
+      remoteDesc = msg;
+      const rejectMessages = [];
+      return that.peerConnection.setLocalDescription(localDesc)
+        .then(() => {
+          that.peerConnection.setRemoteDescription(new RTCSessionDescription(msg));
+        })
+        .then(() => {
+          specBase.remoteDescriptionSet = true;
+          Logger.info('Candidates to be added: ', specBase.remoteCandidates.length,
+            specBase.remoteCandidates);
+          while (specBase.remoteCandidates.length > 0) {
+            // IMPORTANT: preserve ordering of candidates
+            that.peerConnectionFsm.addIceCandidate(specBase.remoteCandidates.shift())
+              .catch(onFsmError.bind(this));
+          }
+          Logger.info('Local candidates to send:', specBase.localCandidates.length);
+          while (specBase.localCandidates.length > 0) {
+            // IMPORTANT: preserve ordering of candidates
+            specBase.callback({ type: 'candidate', candidate: specBase.localCandidates.shift() });
+          }
+        })
+        .catch((error) => {
+          rejectMessages.push(`in: protectedProcessAnswer, error: ${error}`);
+        })
+        .then(() => {
+          firstLocalDescriptionSet = true;
+          firstLocalDescriptionQueue.stopEnqueuing();
+          firstLocalDescriptionQueue.dequeueAll();
+          setTimeout(() => {
+            negotiationQueue.stopEnqueuing();
```

Comment:
why in general we protect calls to next items in `negotiationQueue` and not to `firstLocalDescriptionQueue`?

Reply:
Because... They should also be protected, I'll fix it 👍